schemas/iso19139/src/main/plugin/iso19139/index-fields/index.xsl

```diff
@@ -783,11 +779,10 @@
             <xsl:value-of select="gmd:linkage/gmd:URL"/>
           </xsl:element>
           <link type="object">{
-            "protocol":"<xsl:value-of select="gmd:protocol/*/text()"/>",
-            "url":"<xsl:value-of select="gmd:linkage/gmd:URL"/>",
+            "protocol":"<xsl:value-of select="gn-fn-index:json-escape(gmd:protocol/*/text())"/>",
```

Comment:
As I remember, one has choose not to generate json from xml using xsl,
But to first produce xml before to convert it into json.
Don't have json-escape (also "{" or "}" writing to ouptut) to be postponed at convert xml to json time ?
Producing a mixed file, both json and xml seems to me hazardous.

Reply:
> As I remember, one has choose not to generate json from xml using xsl,

Not sure when we decided that ?




> Producing a mixed file, both json and xml seems to me hazardous.

So far, we are doing it already using Lucene and still have done it like this in the `es` branch eg.
https://github.com/geonetwork/core-geonetwork/blob/es/schemas/iso19139/src/main/plugin/iso19139/index-fields/index.xsl#L766. For now, I don't see better option. The purpose here is to make indexing more robust and at least to fix indexing errors currently triggered by non encoded stuff. Tested on 10K records with Sextant.

Reply:
I was thinking of 
https://github.com/geonetwork/core-geonetwork/pull/2847
Espicially
https://github.com/geonetwork/core-geonetwork/pull/2847/commits/50c193f153d2e790212589fcae9936d7671a25d0
But i am probably out of topic.

Reply:
Indeed I remember your Bolsena experiment in #2847. BTW I don't have resources now to do the indexing in another way. For the time being, the idea was to solve issue due to invalid JSON produced by the process. So we can wait to have more resources to work on that or use that.